### DISCUSSION TITLE
v300 Cluster example

### DISCUSSION THREAD

**Discussion by Tochemey** [ergo-services/ergo#179] [Q&A] [answered]
Hello,

I would like to know whether there is an example on how to use ergo in cluster mode using kubernetes, NATs and co.

**Comment by halturin** [comment]
Hi @Tochemey ,

not sure if i follow your question about the kubernetes. nothing can stop you to start your cluster under the kuber-umbrella. Ergo Framework has service discovery feature embedded. But you may use "Saturn" central registrar as well. It provides you advanced features beside of the nodes registration/discovering - you can subscribe to the registrar events to receive the cluster events (node join/leave, app start/stop). Additionally it provides configuration management feature to broadcast any cluster configuration change on the fly.

for the basic pub/sub you can use "Events" feature of the Ergo Framework. it provides you distributed pub/sub messaging. For the rest - feel free to use NATs, depending on your needs.

**Reply by Tochemey** [reply]
@halturin Thanks for the reply. I was just asking whether there is a need to implement a particular service discovery to use ergo with k8s, NATs and co to let two ergo nodes discover each other in the cluster.

**Reply by halturin** [reply, marked as answer]
no need. it has this feature embedded. similar to Erlang's EPMD.

**Reply by halturin** [reply]
but still, i have a plan to implement kuber's registrar support in the future. not an urgent task, but its in my roadmap.

**Reply by Tochemey** [reply]
> but still, i have a plan to implement kuber's registrar support in the future. not an urgent task, but its in my roadmap.

Nice I will love to contribute

**Reply by halturin** [reply]
just published docs. see https://docs.ergo.services/networking/service-discovering how it works
